packages/vscode-extension/src/exp/index.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+import { IExperimentationService, IExperimentationTelemetry, TargetPopulation, getExperimentationServiceAsync } from "vscode-tas-client";
+import * as pkg from "./../../package.json";
+import { ExtTelemetry } from "./../telemetry/extTelemetry"
+
+export namespace exp {
+    const sharedProperties: {[key: string]: string} = {};
+    let expService: IExperimentationService;
+
+    export function getSharedProperties(): {[key: string]: string} {
+        return sharedProperties;
+    }
+
+    export function getExpService(): IExperimentationService {
+        return expService;
+    }
+
+    export async function initialize(context: vscode.ExtensionContext) {
+        expService = await getExperimentationServiceAsync(
+            pkg.name,
+            pkg.version,
+            TargetPopulation.Public,
+            new ExperimentationTelemetry(),
+            context.globalState
+        );
+    }
+
+    class ExperimentationTelemetry implements IExperimentationTelemetry {
+        public setSharedProperty(name: string, value: string): void {
+            sharedProperties[name] = value;
```

Comment:
These shared properties need to be attached to all telemetry events your extension sends - it's how we can tie an event to a specific experiment group and create a scorecard. It looks like they're just being added to `sharedProperties` and never used?

Reply:
Oh I see - they're being attached to all events sent via `postEvent`. They need to be attached to those events *and* all other events your extension sends.

Reply:
Got it, thanks!